Feature/tvq2mws q view hr to talk

AgnieszkaPawlak opened the pull request:
Create hr view layout.
- Sections: pagination, search, filter, links.

agnieszkazd committed:
Created HR link component.

agnieszkazd committed:
Added tests to HR link component.

agnieszkazd committed:
Added tests to HR container.

agnieszkazd committed:
Fixed format and moved file.

agnieszkazd committed:
Added search context.

agnieszkazd committed:
Added search and filter component.

agnieszkazd committed:
Changed formatting.

agnieszkazd committed:
Updated filter button.

agnieszkazd committed:
Updated search and filter section.

agnieszkazd committed:
Updated package.json

agnieszkazd committed:
Added fetch data and paginate.

agnieszkazd committed:
Updated pagination container.

agnieszkazd committed:
Moved fetch Students to utilis folder.

agnieszkazd committed:
Updated name files.

agnieszkazd committed:
Added import to pagination and deleted unused file.

agnieszkazd committed:
Added hr pages.

agnieszkazd committed:
Fixed formatting

agnieszkazd committed:
Updated styling.

MichalTomczakSE commented on components/hrInfo/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,78 @@
+            {Math.ceil(total)} {/* should be data from BE*/}
+          </span>
+        </div>
+        <ReactPaginate
```
I dont like this pagination, since it looks really diffrent than [here](https://xd.adobe.com/view/864faeb9-d762-4277-a5d1-5b7565dcf543-d31c/screen/faa8be4a-a848-43db-8a13-c7d3e2c1e604/?utm_campaign=WYNIKI+TWOJEGO+ZALICZENIA+MEGAK+-+GRATULACJE&utm_medium=email&utm_source=newsletter&utm_term=2023-04-18). 

If we want stick to that, ill preffer to change some styles of it, maybe add primary-red outline for active, and add some spacing - margins between and larger paddings.

MichalTomczakSE commented on components/hrInfo/Search/searchStudent.tsx:
```diff
@@ -0,0 +1,24 @@
+      <input
+        type="text"
+        className=" w-25/100 bg-navbar-background pl-6"
+        placeholder="szukaj"
```
Add extra padding for this input, spacing between magnifier and placeholder and make first placeholder letter Uppercase.

MichalTomczakSE commented on components/hrInfo/linkHr/linkComponent.tsx:
```diff
@@ -0,0 +1,24 @@
+import Link from "next/link";
+
```
Import next router there

MichalTomczakSE commented on components/hrInfo/linkHr/linkComponent.tsx:
```diff
@@ -0,0 +1,24 @@
+    <ul className="flex gap-x-14 ">
+      <li>
+        <Link
+          className="py-6 focus:border-b-[3px] focus:border-b-secondary-red "
```
Instead of pseudoclass focus, import router and refactor like this: 

```js
 <Link className={router.asPath =="/dashboard/hr/available-students" ? 'border-b-[3px] border-b-secondary-red' : ''  }
 ```
 Because focus works only onclick/after click and border will dissapear, when user clicks on other things.

MichalTomczakSE commented on components/hrInfo/linkHr/linkComponent.tsx:
```diff
@@ -0,0 +1,24 @@
+      </li>
+      <li>
+        <Link
+          className=" py-6 focus:border-b-[3px] focus:border-b-secondary-red"
```
Same as above, but with different path 😄

MichalTomczakSE commented on pages/dashboard/hr/availableStudents.tsx:
```diff
@@ -0,0 +1,9 @@
+const AvailableStudents = () => {
+  return (
+    <>
+      <div> Dostępni kursanci</div>
```
Since this component is going to be same as toTalk, you have choice to copy same structure here, or create Layout, or again add router, and depending on path fetch different data.

MichalTomczakSE commented on pages/dashboard/hr/toTalk.tsx:
```diff
@@ -0,0 +1,58 @@
+  return (
+    <Container>
+      <div className="w-full px-2 pb-5">
+        <Header />
```
This Header component needs to be outside of Container component due to styling. 
```js
<>
<Header />
<Container> 
...
</Container>
<>
```

MichalTomczakSE commented on yarn.lock:
```diff
@@ -653,12 +653,12 @@
 
```
You can delete yarn lock in next commit, due to merge conflict. commit Package json instead.

agnieszkazd committed:
CR fixes pagination

agnieszkazd committed:
CR fixes
- pagination components
- searchStudents styling
- section links styling and switched to next router
- updated view "available students"
- extracting Header component
- added tests

MichalTomczakSE committed:
Merge branch 'develop' into feature/Tvq2mwsQ-view-hr-to-talk

MichalTomczakSE committed:
Delete yarn.lock due to merge conflicts

MichalTomczakSE committed:
Change Button props to match modified interface
